### ISSUE TITLE
Jobs Bank in Collective Distribution

### ISSUE THREAD

**Issue by vrozansky** [givingforum/givingforum#198]
Minnesota (MCF) is planning to launch their new site on our Drupal platform in early November and needs it to include a jobs bank functionality that allows them to accept payments for each job posting. I'm sharing their requirements and ThinkShout's estimate on MCF's behalf. This will be a lengthy description, so please bear with me and provide your comments.

I. MCF REQUIREMENTS

**Data Entry requirements:**

 - Members and nonmembers need an option to submit a job posting form and be able to pay for it (only nonmembers). Only non-members pay for job posting, but the cost is determined by duration of the post (30 days, 45 days, 60 days). Nonmembers get any duration for free. No discount codes.  
 
-Taxonomy tag (or category for a job post) is required. These are: Administration & Finance, Communications & Development, Executive & Senior Management, Grants Management, Programs, Other.  

- Upon successful payment, the job posting should be automatically published. Payee should receive payment confirmation (receipt) reflecting the amount paid, date of payment, possibly last 4 digits of the credit card (not required), and short description. Copy of the email confirmation should be sent to webmaster@mcf.org. It might be nice to allow HTML editor and ability to upload an attachment limited to Word, PDF, and Excel, but there are concerns about security and format consistency. It could be safer to get data in plain text (the actual job description) and let the theme style it. Or, just limit the editor to very few tags (which I know we can do) – bold, italics, and bullets).  

**Fields currently being used:**

-Position
-Organization
-Organization website
-App deadline
-Job Category (choose from a list)
-Job description
-How to Apply

(The following are back-end, not displayed to job seekers):

-Contact First Name
-Contact Last Name
-Contact Phone #
-Contact Email
-Posting Expiration Date

**Back End requirements:**

- Ability to easily see and find all/specific job postings, and publish/unpublish (new view).

- Job posts will expire automatically when either the user-provided deadline or the chosen duration (30/45/60 days) has expired

- Sync to Salesforce as opportunities tied a pre-defined campaign

II. THINKSHOUT'S ESTIMATE

**Workflow:**

- User hits "job posting" product page and fills out fields for the posting. Submitting the form puts the "job posting" in the user's cart.
- User checks out.
- Upon complete payment an email is sent to the purchaser with receipt of purchase and a link to the newly published job posting.
- User logs in and can access a link to edit their job postings from either their profile page or the job posting node page.

**Workflow technical notes:**

- The price of the "job posting" that gets added to the user's cart is determined by their user role and the duration of the posting (field on the job posting form). See DF's nonprofit membership page for an example of variable pricing based on Org budget - https://donorsforum.org/operating-nonprofit-membership.
- "Job posting" type node is automatically created upon successful payment. Possibly use a rule with a custom handler. Be sure to set expiration date of posting.
- Need process to unpublish expired job postings.
- Need permissions to ensure only poster can edit job post nodes they created (IMPORTANT: this assumes only the creator will be able to view submissions, not others at their Org).
- Expose links to job postings submitted by the current user on the user profile paged via user profile form alter.

**ThinkShout Dev todo list:**

- Create a "job posting" product type with specified "currently being used" fields. Estimate: 5 hours.
- Create a "job posting" node type. Configure teaser and full display modes. Teaser will be used on job listing page. Full will have entityform for submitting job application embeded. Estimate: 3 hours.
- Create view for end-user viewing of job postings. Configure facets for filtering by job category. Estimate: 3 hours.
- Rule to create "job posting" node upon successful "job posting" product order payment. Estimate: 5 hours.
- Rule to send email to purchaser with receipt and link to posting upon successful "job posting" product order payment. Estimate: 5 hours.
- Rule run on cron to unpublish expired job postings. Estimate: 5 hours.
- Set permissions so only poster will be able to edit the job posting node. Estimate: 2 hour.
- Create links to jobs posted by the current user on their profile page. Estimate: 5 hours.

III. FORUM's NOTES:

This is a conservative estimate with a total of 33 hours ($5,775) that does not yet include SF integration. To add it will require an additional 5-10 hours. Since we had issues with ThinkShout's estimates in the past, they explicitly inflated this particular estimate to make sure they'd meet it. Still, we recommend budgeting our traditional 30% margin.

ThinkShout also offered to build functionality that will allow job applicants to actually apply for any posted job for an additional 10 hours or so. However, it doesn't look like it's a requirement for MCF or any other RA. If it is, please chime in.

There's a number of additional features that we can build to enhance this jobs board in the coming year, but we recommend concentrating on a simpler Version 1 for now.

To sum up, if all participating RAs decide to partake in the development of a simpler Version 1 that includes SF integration, it'll be approximately $450 each. Please add your comments. Thank you.

**Comment by PhilOhio** [comment]
POH is not interested at this time but maybe in the future.

**Comment by EmilyBaron** [comment]
GWP is not interested at this time.

**Comment by RebekahSeder** [comment]
WRAG might be interested in participating depending on the final cost.

**Comment by flphilanthropy** [comment]
FPN is interested. We don't need an application functionality.

**Comment by mtrimbach** [comment]
CNJG is not interested.

**Comment by kristenruff** [comment]
We might be interested. However, we have fairly strict requirements for our jobs board and even though we communicate those clearly, we still regularly receive job posting announcements that do not qualify. The idea of cutting down on the content management of posting jobs is VERY appealing. However, not being able to approve the jobs before they are posted and the credit card charged might be the barrier for us. I don't want to hold this work up but wondering if anyone else requires an approval process.

**Comment by sgeis** [comment]
Indiana is not interested at this time.

**Comment by vrozansky** [comment]
Thank you to everyone for weighing in. We need to move this work forward, so the deadline for this ticket would be October 9th. In accordance with our MOA, if your RA does not vote on time, it'll be included in the group of RAs interested in that particular ticket.

**Comment by gabriellehana** [comment]
NCG may be interested as well, however we have the same concerns as Kristen, we definitely need to be able to approve the jobs prior to them being charged as we do not allow posts from ineligible parties.

**Comment by wcorkill** [comment]
CMF may also be interested, price will be a consideration since we don't sell to many posts and I do like Kristen's suggestion as well if we were to have this feature.

Thanks again for the reminder Val!

**Comment by Vrylena** [comment]
MPC is not interested at this time.

**Comment by gabriellehana** [comment]
NCG is not interested at this time.

**Comment by AClay2** [comment]
PSW not currently interested in charging for job listings.

**Comment by vrozansky** [comment]
The Forum is in. There is an option to moderate nonmember postings. We can individually customize to even require moderation of member postings as well.

So far we have the following organizations: MCF, Forum, SCG, FPN, WRAG, PNY, NCG, and CMF. Some tentatively opted in pending cost so we'll need definitive votes to finalize individual costs. If we remain with 8 organizations, individual cost will be $688. 10 organizations will reduce it to $550. 15 organizations will further reduce it to $366. If we add to the distro for all participating organizations, individual cost will be $250.

MIE expressed interest in this task as well, but they're going through executive transition and staff turnover period, so their participation maybe delayed.

Please cast your final votes. Thank you.

**Comment by mtrimbach** [comment]
CNJG remains a No.

**Comment by flphilanthropy** [comment]
FPN is still interested, but our participation is on pause due to leadership transition.

**Comment by canheier** [comment]
ncg - yes

**Comment by PhilOhio** [comment]
POH is interested.

**Comment by vrozansky** [comment]
We'll be closing voting on this ticket by COB February 5th. If you haven't voted yet, please do so now. Thank you.

**Comment by kristenruff** [comment]
We already voted but to confirm, PNY yes.

**Comment by kcifone** [comment]
CCP is not interested at this time, but may be in the future

**Comment by vrozansky** [comment]
FYI - RAs that have jobs bank installed just received an additional field called Add File that allows users to upload their job posting via an attachment as well. If anyone needs a custom configuration, please contact me directly by email. Thank you.
